test/distributed/partition_kernels.cpp

```diff
@@ -69,6 +69,9 @@ class Partition : public CommonTestFixture {
             gko::make_array_view(
                 this->exec, dpart->get_num_parts(),
                 const_cast<local_index_type*>(dpart->get_part_sizes())));
+
+        GKO_ASSERT_SEGMENTED_ARRAY_EQ(part->get_ranges_by_part(),
```

Comment:
Side-node: Since this is a function returning `void`, this will not prevent subsequent operations from executing.

Reply:
This should be implemented the same way as the `GKO_ASSERT_ARRAY_EQ` macro. The `segmented_array_equal` function will return something, so I don't know if there is anything more to do here.

Reply:
I mean the overall approach of implementing a custom `assert_something` function may cause tests to continue executing after the assertion fails, which may or may not be desirable. It's fine here, but something to keep in mind. To fix this, we would need to return `testing::AssertionResult` and wrap the call in a macro. But nothing to hold back the code here, just a comment that people should be aware of.